platform-mbean/src/main/java/org/jboss/as/platform/mbean/AbstractExtendedMBean.java

```
@@ -0,0 +1,116 @@
+        Objects.requireNonNull(attributeName);
+        try {
+            for (MBeanAttributeInfo attribute
+                    : ManagementFactory.getPlatformMBeanServer().getMBeanInfo(name).getAttributes()) {
+                if (attributeName.equals(attribute.getName())) {
+                    return true;
+                }
+            }
+        } catch (InstanceNotFoundException | IntrospectionException | ReflectionException ex) {
+            // XXX OK, attribute not available.
+        }
+        return false;
+    }
+
+    // Operation
+    boolean isOperationDefined(String operationName, String[] types) {
+        Objects.requireNonNull(operationName);
+        Objects.requireNonNull(types);
+        try {
+            for (MBeanOperationInfo op
+                    : ManagementFactory.getPlatformMBeanServer().getMBeanInfo(name).getOperations()) {
+                if (operationName.equals(op.getName())) {
+                    List<String> signature = new ArrayList<>();
+                    for(MBeanParameterInfo p : op.getSignature()) {
+                        signature.add(p.getType());
+                    }
+                    List<String> receivedTypes =  Arrays.asList(types);
+                    if(receivedTypes.equals(signature)) {
+                        return true;
+                    }
+                }
+            }
+        } catch (InstanceNotFoundException | IntrospectionException | ReflectionException ex) {
+            // XXX OK, attribute not available.
+        }
+        return false;
+    }
+
+    Object getAttribute(String attributeName) throws OperationFailedException {
+        try {
+            return ManagementFactory.getPlatformMBeanServer().
+                    getAttribute(name, attributeName);
+        } catch (AttributeNotFoundException
+                | InstanceNotFoundException
+                | MBeanException
+                | ReflectionException ex) {
+            throw new OperationFailedException(ex);
```

This comment applies to this commit overall; I just randomly picked this line.

We should only throw OFE or any other exception type that implements OperationClientException if the failure indicates an end user mistake. For any other cases, an OperationStepHandler should throw a runtime exception of some sort.

The server kernel exception handling relies on this, i.e. perhaps dealing with user mistakes one way while handling other failures differently.